cmd/k-license/k-license_test.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"testing"
+)
+
+var optTest = &Options{}
+
+// Running E2E Test
+func TestAddLicense(t *testing.T) {
+	optTest.templatesDir = "../../hack/boilerplate"
+	optTest.excludeDirs = excludeDirsLocations
+	optTest.path = "./testdata"
+	optTest.confirm = false
+	optTest.Run()
+}
+
+func TestAddConfirmLicense(t *testing.T) {
+	optTest.templatesDir = "../../hack/boilerplate"
+	optTest.excludeDirs = excludeDirsLocations
+	optTest.path = "./testdata"
+	optTest.confirm = true
+	optTest.Run()
+}
```

Comment:
This still isn't testing what we want:
1. Need to verify if err returned is nil or not
2. Need to confirm if the files actually change or not based on `confirm`.
Something like:
```go
func TestAddLicense(t *testing.T) {
	cases := []struct {
		confirm bool
		desc    string
	}{
		{
			confirm: false,
			desc:    "confirm=true, files should actually change",
		},
		{
			confirm: true,
			desc:    "confirm=false, files should not actually change",
		},
	}
	for _, c := range cases {
		t.Run(c.desc, func(t *testing.T) {
			optTest.templatesDir = "../../hack/boilerplate"
			optTest.excludeDirs = excludeDirsLocations
			optTest.confirm = c.confirm
			// create a tmp dir with the test data that we can delete and
			// recreate after each test case.
			tmpTestDir := createTmpDir()
			optTest.path = tmpTestDir
			if err := optTest.Run(); err != nil {
				t.Errorf("error running test case %s: %v", c.desc, err)
			}
		})
	}
}
```